pennylane/transforms/dynamic_one_shot.py

```diff
@@ -237,106 +229,144 @@ def init_auxiliary_tape(circuit: qml.tape.QuantumScript):
     )
 
 
-# pylint: disable=too-many-branches
+def _measurement_with_no_shots(measurement):
+    return (
+        np.nan * np.ones_like(measurement.eigvals())
+        if isinstance(measurement, ProbabilityMP)
+        else np.nan
+    )
+
+
+def _get_is_valid_has_valid(mcm_samples, all_mcms, interface):
+    # Can't use boolean dtype array with tf, hence why conditionally setting items to 0 or 1
+    has_postselect = qml.math.array(
+        [[op.postselect is not None for op in all_mcms]],
+        like=interface,
+        dtype=mcm_samples.dtype,
+    )
+    postselect = qml.math.array(
+        [[0 if op.postselect is None else op.postselect for op in all_mcms]],
+        like=interface,
+        dtype=mcm_samples.dtype,
+    )
+    is_valid = qml.math.all(mcm_samples * has_postselect == postselect, axis=1)
+    has_valid = qml.math.any(is_valid)
+    return is_valid, has_valid
+
+
+# pylint: disable=unused-argument
 def parse_native_mid_circuit_measurements(
     circuit: qml.tape.QuantumScript,
-    aux_tapes: qml.tape.QuantumScript,
-    results: TensorLike,
+    _removed_arg=None,  # need to not break catalyst
```

Comment:
Catalyst uses this function and was still passing `aux_tapes`.  But we really didn't need it here.